src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs

```diff
@@ -115,15 +115,18 @@ private static int FindIndexOfAny(string format, char[] chars, int startIndex, i
 
         public string Format(object[] values)
         {
+            object[] formattedValues = null;
+
             if (values != null)
```

Comment:
Might values be Length == 0? There's no benefit allocating in that case.

Reply:
No, when the values array is either null or of length 0, the [Log formatter is set to null](https://github.com/dotnet/runtime/blob/ff1ea091265968e218664aee72a081965945e18a/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/FormattedLogValues.cs#L32-L53) and in that case, array formatting [is skipped](https://github.com/dotnet/runtime/blob/ff1ea091265968e218664aee72a081965945e18a/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/FormattedLogValues.cs#L98-L106).

This does makes the check for a null array redundant, no?